modules/gateways/blockonomics.php

```diff
@@ -265,53 +242,106 @@ function doTest() {
                     newBtn.disabled = true;
-                    Object.keys(CELLS).forEach(crypto => {
-                        let cell = CELLS[crypto]
-                        cell.innerHTML = "$trans_text_testing";
-                    })
+                    responseDiv.innerHTML = "$trans_text_testing";
                     oReq.send();
-                    
                 }
-
-			}
+            }
 
             // For Non AJAX Based Submission
             if(sessionStorage.getItem("runTest") && !document.querySelector('#manage .errorbox')) {
                 sessionStorage.removeItem("runTest");
                 doTest()
             }
 
+            /**
+             * Check API key and update store name
+             */
+            const apiKeyInput = document.getElementsByName('field[ApiKey]')[0];
+            const storeNameField = document.getElementsByClassName('store-name-description')[0];
+
+            function updateStoreName() {
+                if (!apiKeyInput.value) {
+                    storeNameField.innerHTML = 'Enter API key to fetch store name';
+                    return;
+                }
+
+                // Show loading message
+                storeNameField.innerHTML = 'Loading store name...';
+
+                // Use local proxy to avoid CORS issues
+                var storesetupUrl = "$system_url" + "modules/gateways/blockonomics/storesetup.php";
+                fetch(storesetupUrl)
+                    .then(response => response.json())
+                    .then(response => {
+                        if (response.needs_store) {
+                            storeNameField.innerHTML = 'No stores found for this API key';
+                            return;
+                        }
+
+                        if (response.success && response.store_name) {
+                            storeNameField.innerHTML = response.store_name;
+                        } else {
+                            storeNameField.innerHTML = 'No matching store found for this callback URL';
+                        }
+                    })
+                    .catch(error => {
+                        storeNameField.innerHTML = 'Error fetching store information';
+                        console.error('Store fetch error:', error);
+                    });
+            }
+
+            updateStoreName();
+
+            // Update store name
+            if(typeof jQuery != 'undefined') {
+                jQuery(document).on('ajaxComplete', updateStoreName)
+            }
+            // Add onblur event listener
+            apiKeyInput.addEventListener('blur', () => {
+                // Submit the form
+                if(saveButtonCell.querySelector('button[type=submit]')){
+                    saveButtonCell.querySelector('button[type=submit]').click();
+                } else {
+                    saveButtonCell.querySelector('input[type=submit]').click();
+                }
+            });
```

Comment:
API key is saved every time it's changed because of the `onblur` event listener that automatically triggers form submission. This means even if we don't manually click "Save Changes" or "Test Setup", the API key is saved when you:
1. Click outside the API key field
2. Switch to another tab/window
3. Move focus to another field

This is potentially problematic from a UX perspective because:
1. It saves changes without explicit user consent
2. Users might be testing different API keys but not intend to save them
3. There's no confirmation or undo mechanism

Lets avoid this.